<a href="https://colab.research.google.com/github/andrewbilbrey/Billboard-Song-Data-Assignment-6-/blob/master/assignments/Assignment_9.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
import math
import scipy.stats as stats
import numpy as np

### Step 1. Calculate Sample Size and Sample Mean

In [3]:
x = [6414, 1550, 2109, 9350, 21828, 4300, 5944, 5722, 2825, 2044,
5481, 5200, 5853, 2750, 10012, 6357, 27000, 9414, 7681, 3200,
17500, 9200, 7380, 18314, 6557, 13713, 17768, 7493, 2771, 2861,
1263, 7285, 28165, 5080, 11622]
n = len(x)
print("Sample size =", n)

Sample size = 35


In [4]:
sample_mean = round(np.mean(x),2)
print("Sample mean =", sample_mean)

Sample mean = 8628.74


Point estimate = 8628.74

#Step 2. Calculate the Sample Standard Deviation & Sample Error

In [7]:
sample_std = round(np.std(x, ddof=1),2)         
print("Sample Standard Deviation =", sample_std)

Sample Standard Deviation = 6943.88


In [8]:
std_err = sample_std / math.sqrt(len(x))
std_err = round(std_err,2)
print("Sample Standard Error is", std_err)

Sample Standard Error is 1173.73


#Step 3. Calculate t Critical Value using t-Distribution

In [9]:
t_value = stats.t.ppf(0.975, 15)
print("t critical value = ", round(t_value,2))

t critical value =  2.13


# Step 4. Calculate Margin of Error

In [10]:
margin_of_error = round(t_value * std_err,2) 
print("Margin of Error =", margin_of_error)

Margin of Error = 2501.75


# Step 5. Calculate Lower and Upper Limit of the Confidence Interval

In [11]:
lower_limit = sample_mean - margin_of_error
print ("Lower Limit = ", lower_limit)

Lower Limit =  6126.99


In [12]:
upper_limit = sample_mean + margin_of_error
print ("Upper Limit = ", upper_limit)

Upper Limit =  11130.49


#Step 6. Now you have the confidence interval

In [13]:
print("The 95% Confidence Interval = (", lower_limit, ",", upper_limit, ")")

The 95% Confidence Interval = ( 6126.99 , 11130.49 )
